crates/topos-sequencer-subnet-runtime/src/proxy.rs

```diff
@@ -122,11 +122,17 @@ impl SubnetRuntimeProxy {
                                     "Source head certificate id received {:?}",
                                     certificate_and_position
                                 );
-                                // If the position is not provided, it should start form -1, so that first fetched is subnet genesis block
+                                // If tce source head position is provided, continue syncing from it
+                                // If `start_block` sequencer parameter is provided and tce source head is missing,
+                                // we should start syncing from that block instead of genesis
+                                // If both tce source head position and start_block parameter are not provided,
+                                // sync should start form -1, so that first fetched is subnet genesis block
+                                let default_block_sync: i128 =
+                                    config.start_block.map(|i| i as i128).unwrap_or(-1);
```

Comment:
Better to move it one scope above in order to set `latest_acquired_subnet_block_number` [here](https://github.com/topos-protocol/topos/pull/318/files#diff-7f258e647b37319822e9ed2541297b6800806a063fa7667e60da9806ef72e891R107) as well which is still set to `-1`

Reply:
Currently it does not change flow (it would panic if source head could not be retrieved), but I agree, it should be moved up